src/openrct2/core/Numerics.hpp

```diff
@@ -64,6 +65,27 @@ namespace OpenRCT2::Numerics
 #endif
     }
 
+    [[maybe_unused]] constexpr bool isPowerOf2(int v)
+    {
+        return v && ((v & (v - 1)) == 0);
+    }
+
+    // Rounds an integer down to the given power of 2. y must be a power of 2.
+    [[maybe_unused]] constexpr int floor2(const int x, const int y)
+    {
+        if (!isPowerOf2(y))
+            throw std::logic_error("floor2 should only operate on power of 2");
```

Comment:
This should be an assert, having this checked all the time is a waste of cycles.